### ISSUE TITLE
maintain history of online status 

### ISSUE THREAD

**Issue by btry** [flyve-mdm/flyve-mdm-glpi#34]
This allows to compute availability statistics (ie: 56% online) and graph it (per device, per fleet or globally?)

**Comment by DIOHz0r** [comment]
Hello, I'm working on the issue but with a different graph.

**Comment by btry** [comment]
Hi

I only suggested graphs. If you have in mind some graphs, please tell about them.

**Comment by DIOHz0r** [comment]
My plan is to show a graph like this with the days as X values and hours for Y...
![imagen](https://user-images.githubusercontent.com/1426313/35829366-07f7821e-0a99-11e8-97ac-3eba90af0fb0.png)

**Comment by btry** [comment]
Hours as Y : do you mean a vertical axis from 0 to 24 ? or is Y the duration of online status of a device ?

**Comment by DIOHz0r** [comment]
Yes, a daily range where each point is a ping received from each device.

**Comment by btry** [comment]
Hummm I think this is not a very good approach to rely on pings.

Remember, the agent now connect to Mosquitto with a will. The will is saved by Mosquitto, and sent to subscribers on behalf on the agent. The will is designed to tell the backend when an agent times out.

The ping is less accurate than the will, except if you setup a ping every minute (which consumes bandiwdth and power). 

I think you need to log connection and loss of connection events. If possible, use the Log itemtype in GLPI, and build a query to collect these events.
